packages/x-codemod/src/v8.0.0/charts/rename-sparkline-colors-to-color/expected.spec.tsx

```diff
@@ -0,0 +1,19 @@
+/* eslint-disable no-restricted-imports */
+import * as React from 'react';
+import { SparkLineChart } from '@mui/x-charts';
+
+const data = [1, 2];
+
+function Chart() {
+  const fn = (mode) => (mode === 'light' ? ['black'] : ['white']);
+
+  // prettier-ignore
+  return (
+    (<React.Fragment>
+      <SparkLineChart data={data} color={['red'][0]} />
+      <SparkLineChart data={data} color={typeof fn === "function" ? mode => fn(mode)?.[0] : fn} />
```

Comment:
Not sure if this is a valid approach. 

This introduces a slight runtime performance hit because, when codemodding, we can't guarantee whether `fn` is a function or an array. 

If you think this or another example are too complex, I can just remove this and leave this as an unhandled case.